### PR TITLE
dev-ruby/rdoc: fix shebang for gentoo prefix

### PR DESCRIPTION
**Pull request by littlewu2508**
Package-Manager: Portage-3.0.18, Repoman-3.0.3
Signed-off-by: Yiyang Wu <xgreenlandforwyy@gmail.com>

### CODE REVIEW

File: dev-ruby/rdoc/rdoc-6.1.2.ebuild [hunk 1]
```diff
@@ -82,6 +82,7 @@ all_ruby_install() {
 				ruby_fakegem_binwrapper $bin /usr/bin/${bin}${version}
 				sed -i -e "1s/env ruby/ruby${version}/" \
 					"${ED}/usr/bin/${bin}${version}" || die
+				if use prefix; then hprefixify ${ED}/usr/bin/${bin}${version}; fi
```
Comment (graaff) [inline dev-ruby/rdoc/rdoc-6.1.2.ebuild:85]: The argument to hprefixify needs to be quoted here, since ${ED} may contain spaces.

**Comment by graaff** [comment]
The rdoc ebuilds did already change a bit so this commit did no longer apply as-is. I have added your change to the 6.3.2 ebuild in 8c3c4b660050062f63f964abd75e83f39ea92136, but using the more compact "use prefix &&" syntax.

Thanks for the fix!

**Comment by heroxbd** [comment]
This patch back fires with `ruby-fakegem.eclass`.  The line 
```bash
rubycmd="$(ruby_implementation_command ${implementation})"
```
finds a ruby implementation inside a Prefix, not needed to be prefixified again.

**Comment by thesamesam** [comment]
> This patch back fires with `ruby-fakegem.eclass`. The line
> 
> ```shell
> rubycmd="$(ruby_implementation_command ${implementation})"
> ```
> 
> finds a ruby implementation inside a Prefix, not needed to be prefixified again.

Actually, thinking about it, probably better to just open a new PR. Opening the old one is easier to miss (and confusing as there's no patch to be applied). It's more of a bug report.

**Comment by heroxbd** [comment]
> Actually, thinking about it, probably better to just open a new PR. Opening the old one is easier to miss (and confusing as there's no patch to be applied). It's more of a bug report.

Sure, I have come up with a revised patch.  Will open another PR.

**Pull request closed without merging**